tools/autograd/gen_python_functions.py

```diff
@@ -403,6 +405,13 @@ def unpack_variable(name, unpack_expr, typename):
                 """.format(name=name, expr=unpack_expr, typ='DimnameList')
                 return [line.strip() for line in result.split('\n')]
 
+            if typename == 'c10::optional<IntArrayRef>':
+                result = """\
+                    auto __{name} = {expr};
+                    c10::optional<{typ}> {name} = __{name} ? c10::make_optional({typ}(__{name}.value())) : c10::nullopt;
```

Comment:
This is wrong. You're taking out a reference to `value()` but `__{name}.value()` is a temporary and will die once you exit this scope. That's why you're getting garbage.

Reply:
Ok I see. According to a comment just above this code:
```
# optional<ArrayRef<T>>. One needs to unwrap the optional and rewrap.
```
I'm not sure how to do that.